src/com/pensio/api/PensioMerchantAPI.java

```diff
@@ -288,6 +289,21 @@ protected void setPaymentRequestParameters(
 		addPaymentInfo(paymentRequest, params);
 
 		addOrderLines("orderLines",params, paymentRequest.getOrderLines());
+
+		if(paymentRequest.getAgreementConfig() != null)
+		{
+			addParam(params, "agreement[id]", paymentRequest.getAgreementConfig().getAgreementId());
+			addParam(params, "agreement[type]", paymentRequest.getAgreementConfig().getAgreementType().name());
+			addParam(params, "agreement[unscheduled_type]", paymentRequest.getAgreementConfig().getAgreementUnscheduledType().name());
+			addParam(params, "agreement[expiry]", DateHelper.formatDate("yyyyMMdd", paymentRequest.getAgreementConfig().getAgreementExpiry()));
+			addParam(params, "agreement[frequency]", paymentRequest.getAgreementConfig().getAgreementFrequency());
+			addParam(params, "agreement[next_charge_date]", DateHelper.formatDate("yyyyMMdd", paymentRequest.getAgreementConfig().getAgreementNextChargeDate()));
+			addParam(params, "agreement[admin_url]", paymentRequest.getAgreementConfig().getAgreementAdminUrl());
+		}
+		//below check is to be removed as getAgreementType() is deprecated.
+		if(paymentRequest.getAgreementType() != null){
+			addParam(params, "agreement[type]", paymentRequest.getAgreementConfig().getAgreementType().name());
+		}
```

Comment:
1. removing deprecated check
2. checking if `getAgreementConfig().getAgreementType()` is not null, as that's what's actually used as parameter value
3. 
```suggestion
		if(paymentRequest.getAgreementConfig().getAgreementType() != null){
			addParam(params, "agreement[type]", paymentRequest.getAgreementConfig().getAgreementType().name());
		}
```

Reply:
The check was placed so that if someone uses the deprecated function and not the new one to set the value, it should get sent to the gateway.
All values from AgreementConfig are already being sent on [line 293](https://github.com/AltaPay/sdk-java/blob/ad67db5356f5d113c2c76ce091e63987818871b9/src/com/pensio/api/PensioMerchantAPI.java#L293) above the check under discussion.